monai/transforms/post/dictionary.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2020 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+A collection of dictionary-based wrappers around the "vanilla" transforms for model output tensors
+defined in :py:class:`monai.transforms.utility.array`.
+
+Class names are ended with 'd' to denote dictionary-based transforms.
+"""
+
+from monai.utils.misc import ensure_tuple_rep
+from monai.transforms.compose import MapTransform
+from monai.transforms.post.array import SplitChannel
+
+
+class SplitChanneld(MapTransform):
+    """
+    Dictionary-based wrapper of :py:class:`monai.transforms.SplitChannel`.
+    All the input specified by `keys` should be splitted into same count of data.
+
+    """
+    def __init__(self, keys, output_postfixes, to_onehot=False, num_classes=None):
+        """
+        Args:
+            keys (hashable items): keys of the corresponding items to be transformed.
+                See also: :py:class:`monai.transforms.compose.MapTransform`
+            output_postfixes (list, tuple): the postfixes to construct keys to store splitted data.
+                for example: if the key of input data is `pred` and split 2 classes, the output
+                data keys will be: pred_(output_postfixes[0]), pred_(output_postfixes[1])
+            to_onehot (bool or list of bool): whether to convert the data to One-Hot format, default is False.
+        num_classes (int or list of int): the class number used to convert to One-Hot format if `to_onehot` is True.
+        """
+        super().__init__(keys)
+        assert isinstance(output_postfixes, (list, tuple)), 'must specify key postfixes to store splitted data.'
```

Comment:
Here we should be checking the condition and raising a ValueError instead of using assert.

Reply:
updated in the latest patch.
Thanks.